Core/include/Acts/TrackFitting/detail/MergeGaussianMixture.hpp

```diff
@@ -195,20 +196,150 @@ auto gaussianMixtureMeanCov(const components_t components,
+                        const angle_desc_t &desc = angle_desc_t{}) {
+  constexpr int D = []() {
+    using ParsType = decltype(unevaluated_context_centralValue(
+        std::declval<components_t>(), std::declval<projector_t>()));
+    return ParsType::RowsAtCompileTime;
+  }();
+
+  // Lambda used to correct cyclic coordinates in the computation
+  auto cyclicDiff = [](auto d, auto &diff, const auto &a, const auto &b) {
+    diff[d.idx] = difference_periodic(a[d.idx] / d.constant,
+                                      b[d.idx] / d.constant, 2 * M_PI) *
+                  d.constant;
+  };
+
+  // Compute the value of the pdf of a single multivariate gaussian
+  auto single_pdf = [&](const auto &x, const auto &centralValue,
+                        const auto &cov) {
+    const auto a = 1.0 / std::sqrt(std::pow(2 * M_PI, D) * cov.determinant());
+    ActsVector<D> r = x - centralValue;
+    std::apply([&](auto... d) { (cyclicDiff(d, r, x, centralValue), ...); },
+               desc);
+    const auto b = -0.5 * (x - centralValue).transpose() * cov.inverse() *
+                   (x - centralValue);
+    return a * std::exp(b);
+  };
+
+  // Compute the value of the gaussian mixture pdf at x
+  auto mixture_pdf = [&](const auto &x) {
+    double res = 0.0;
+
+    for (const auto &cmp : components) {
+      const auto &[weight, centralValue, cov] = proj(cmp);
+      res += weight * single_pdf(x, centralValue, cov);
+    }
+
+    return res;
+  };
+
+  // The fixed-point equation (see ref)
+  auto f = [&](const auto &x) {
+    const auto p_x = mixture_pdf(x);
+
+    ActsSquareMatrix<D> a = ActsSquareMatrix<D>::Zero();
+    ActsVector<D> b = ActsVector<D>::Zero();
+
+    for (const auto &cmp : components) {
+      const auto &[weight, centralValue, cov] = proj(cmp);
+      const auto p_m_x = weight * single_pdf(x, centralValue, cov) / p_x;
+
+      a += p_m_x * cov.inverse();
+      b += p_m_x * (cov.inverse() * centralValue);
+    }
+
+    return a.inverse() * b;
+  };
+
+  // Compute the hessian of a gaussian mixture at x
+  auto hessian = [&](const auto &x) {
+    ActsSquareMatrix<D> h = ActsSquareMatrix<D>::Zero();
+
+    for (const auto &cmp : components) {
+      const auto &[weight, centralValue, cov] = proj(cmp);
+      const auto a = weight * single_pdf(x, centralValue, cov);
+      ActsVector<D> r = x - centralValue;
+      std::apply([&, &centralValue = centralValue](
+                     auto... d) { (cyclicDiff(d, r, x, centralValue), ...); },
+                 desc);
+      const auto b = (x - centralValue) * (x - centralValue).transpose() - cov;
+      h += a * (cov.inverse() * b * cov.inverse());
+    }
+
+    return h;
+  };
+
+  // We only store the highest mode
+  // TODO Replace with Acts::Result
+  std::optional<ActsVector<D>> mode;
+  double mode_val = 0;
+
+  // Algorithm parameters
+  constexpr double tol = 1.e-8;
+  constexpr double eigv_max = 0.01;
+  constexpr int iter_max = 50;
```

Comment:
are these params stable/optimal enough to be used in all scenarios?

Reply:
Hmm so it was quite a while ago that I implemented this, and I'm actually not totally sure. But since I test several mixture configurations in the unit test, it can't be too bad... But it could be certainly a performance optimization to tune these parameters...

Another point: Currently the interface (an `enum`) wouldn't allow for configuration anyways, so this would have to be changed to a functor-based interface (which would be good...). However, I definitely would to it in a separate PR.